File: AmongNeewAmigosBackEnd/src/main/java/leandromangieri/portfolio/authentication/dto/SignupRequestAmong.java
package leandromangieri.portfolio.authentication.dto;


import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.*;

@Data
@NoArgsConstructor
public class SignupRequestAmong {
    @Email
    private String email;

    @NotEmpty(message = "password cannot be empty")
    @Size(min = 8, message = "Tamanho mínimo para o password são 8 caracteres")
    private String password;

    @NotEmpty(message = "nickName cannot be empty")
    private String nickName;
}

File: AmongNeewAmigosFrontEnd/AmongNeewAmigosFrontEnd/src/app/authentication/dtos/SignupRequestAmong.ts
export class SignupRequestAmong {
    constructor(email : string, password : string, nickName : string) {
        this.email = email;
        this.password = password;
        this.nickName = nickName;
    }

    email  : string;
    password  : string;
    nickName  : string;
}